hooks/post_gen_project.py

```diff
@@ -15,25 +15,35 @@ def clean_unused_template_settings():
     working = os.path.abspath(os.path.join(os.path.curdir))
     templates = os.path.join(working, '{{cookiecutter.repo_name}}', 'templates')
 
-    layout = None
-    mytemplate = None
     removal_text = None
-    view = os.path.join(working, '{{cookiecutter.repo_name}}', 'views.py')
+
+    templ_to_remove = set([
+        'layout.pt', 'mytemplate.pt',
+        'layout.jinja2', 'mytemplate.jinja2',
+        'layout.mako', 'mytemplate.mako',
+    ])
```

Comment:
Looking at this and the previous version, this gives me another idea.

```python
language_extensions = {'chameleon': '.pt', 'jinja': '.jinja2', 'mako': '.mako'}
removals = del language_extensions[selected_lang]
```

Now you've got an iterable that consists of the things to remove, where the keys are the languages, and the values are the template extensions. So instead of the if/elif construct where you must carefully define what to remove in each block, use a function instead.